src/games/performance/Perf.java

```diff
@@ -0,0 +1,62 @@
+package games.performance;
+
+import java.util.prefs.Preferences;
+
+
+/**
+ * Provides a high level API to the game engine for performance measurements.
+ * This class handles the library details and sends output to 'PerformanceConsole.java'
+ */
+public class Perf {
+
+  private static final String LOG_PERFORMANCE_KEY = "logPerformance";
+  private static boolean enabled;
+
+
+  static {
+    enabled = isEnabled();
+    if (enabled) {
+      PerformanceConsole.getInstance().setVisible(true);
+    }
+  }
+
+  public static void setEnabled(final boolean isEnabled) {
+    if (enabled != isEnabled) {
+      enabled = isEnabled;
+      PerformanceConsole.getInstance().setVisible(enabled);
+      storeEnabledPreference();
+    }
+  }
+
+  private static void storeEnabledPreference() {
+    final Preferences prefs = Preferences.userNodeForPackage(EnablePerformanceLoggingCheckBox.class);
+    prefs.put(LOG_PERFORMANCE_KEY, Boolean.valueOf(enabled).toString());
+  }
+
+  public static boolean isEnabled() {
+    final Preferences prefs = Preferences.userNodeForPackage(EnablePerformanceLoggingCheckBox.class);
+    return prefs.getBoolean(LOG_PERFORMANCE_KEY, false);
+  }
+
+  public static PerfTimer startTimer(Object obj, String title) {
```

Comment:
I've been working on another branch that builds on this, uses this as infrastructure. We've had multiple needs for being able to time things easily. Each time I add an ad-hoc class that is similar to StopWatch that has a simpler API and then delete it afterwards. Having a more advanced framework will help.

But... it does look like that branch did get dropped.. My mistake. Regardless, there are some timers I certainly wouldn't mind adding in the short term. With recent work on downloading maps, map parsing, game loading, having this framework will be of use very quickly.